apps/web/src/utils/hooks/useVariant.tsx

```diff
@@ -0,0 +1,31 @@
+import { useEffect, useState } from 'react';
+import { Experiment } from '@amplitude/experiment-js-client';
+
+import { isDevelopment } from 'apps/web/src/constants';
+
+const defaultDeploymentKey = {
+  development: 'client-Wvf63OdaukDZyCBtwgbOvHgGTuASBZFG',
+  production: '',
+};
+
+export default function useVariant(
+  flagKey: string,
+  deploymentKey = isDevelopment
+    ? defaultDeploymentKey.development
+    : defaultDeploymentKey.production,
+): string {
+  const [variant, setVariant] = useState('');
+
+  useEffect(() => {
+    try {
+      const experiment = Experiment.initialize(deploymentKey);
```

Comment:
Hm, I don’t think we’ll want to initialize this every time we want to use this hook. I think ideally we would reuse the one experiment client that we’ve initialized previously.